samples/apps/copilot-chat-app/webapp/src/libs/semantic-kernel/sk-utilities.ts

```diff
@@ -0,0 +1,48 @@
+import { IPlan, IPlanInput } from '../models/Plan';
+
+export const isPlan = (object: string) => {
+    // backslash has to be escaped since it's a JSON string
+    // eslint-disable-next-line
+    const planPrefix = `proposedPlan\":`;
+    return object.indexOf(planPrefix) !== -1;
+};
+
+export const parsePlan = (response: string): IPlan | null => {
+    if (isPlan(response)) {
+        const parsedResponse = JSON.parse(response);
+        const plan = parsedResponse.proposedPlan;
+        const userIntentPrefix = 'User intent: ';
+        const index = plan.description.indexOf(userIntentPrefix);
+
+        return {
+            userIntent: plan.description,
+            description: index !== -1 ? plan.description.substring(index + userIntentPrefix.length).trim() : '',
+            steps: extractPlanSteps(plan),
+        };
+    }
+    return null;
+};
+
+const extractPlanSteps = (plan: any) => {
+    const planInputs: IPlanInput[] = [];
+    for (var input in plan.state) {
+        if (
+            // Omit reserved context variable names
+            plan.state[input].Key !== 'INPUT' &&
```

Comment:
Can we store the list of reserved keywords somewhere, and have a helper function do the check. I have a feeling as we interact with plans more, adding to context variables is going to happen in more places. It'll be easier to keep a track of it if its all in one place.